components/src/dynamo/common/backend/logprobs.py

```
@@ -0,0 +1,265 @@
+            selected = next(iter(token_logprobs_dict.values()), None)
+            if selected is None:
+                continue
+        log_probs.append(float(selected.logprob))
+
+        position_entries: list[dict[str, Any]] = []
+        for tok_id, info in token_logprobs_dict.items():
+            token_str = getattr(info, "decoded_token", None)
+            if not token_str and tokenizer is not None:
+                try:
+                    token_str = tokenizer.decode([tok_id])
+                except Exception:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)